hpat/hiframes_typed.py

```diff
@@ -751,6 +754,96 @@ def f(arr, on_arr, w, center):  # pragma: no cover
+        """
+        func = guard(get_definition, self.func_ir, rhs.args[1])
+        if func is None or not (isinstance(func, ir.Expr)
+                                and func.op == 'make_function'):
+            raise ValueError("lambda for combine not found")
+
+        out_typ = self.typemap[lhs.name].dtype
+
+        # If we are called with 3 arguments, we must use 3rd arg as a fill value,
+        # instead of Nan.
+        use_nan = len(rhs.args) == 2
+
+        # prange func to inline
+        if use_nan:
+            func_text = "def f(A, B):\n"
+        else:
+            func_text = "def f(A, B, C):\n"
+        func_text += "  numba.parfor.init_prange()\n"
+        func_text += "  n1 = len(A)\n"
+        func_text += "  n2 = len(B)\n"
+        func_text += "  n = n1\n"
+        func_text += "  if n2 > n:\n"
+        func_text += "      n = n2\n"
+        func_text += "  S = numba.unsafe.ndarray.empty_inferred((n,))\n"
+        func_text += "  for i in numba.parfor.internal_prange(n):\n"
+        if use_nan:
+            func_text += "    t1 = np.nan\n"
+        else:
+            func_text += "    t1 = C\n"
+        func_text += "    if i < n1:\n"
+        func_text += "      t1 = A[i]\n"
+        if use_nan:
+            func_text += "    t2 = np.nan\n"
+        else:
+            func_text += "    t2 = C\n"
+        func_text += "    if i < n2:\n"
+        func_text += "      t2 = B[i]\n"
+        func_text += "    S[i] = map_func(t1, t2)\n"
+        if out_typ == hpat.pd_timestamp_ext.datetime_date_type:
+            func_text += "  ret = hpat.hiframes_api.to_date_series_type(S)\n"
+        else:
+            func_text += "  ret = S\n"
+        func_text += "  return ret\n"
+
+        loc_vars = {}
+        exec(func_text, {}, loc_vars)
+        f = loc_vars['f']
+
+        _globals = self.func_ir.func_id.func.__globals__
+        f_ir = compile_to_numba_ir(f, {'numba': numba, 'np': np, 'hpat': hpat})
+
+        # fix definitions to enable finding sentinel
+        f_ir._definitions = build_definitions(f_ir.blocks)
+        topo_order = find_topo_order(f_ir.blocks)
+
+        # find sentinel function and replace with user func
+        for l in topo_order:
+            block = f_ir.blocks[l]
+            for i, stmt in enumerate(block.body):
+                if (isinstance(stmt, ir.Assign)
+                        and isinstance(stmt.value, ir.Expr)
+                        and stmt.value.op == 'call'):
+                    fdef = guard(get_definition, f_ir, stmt.value.func)
+                    if isinstance(fdef, ir.Global) and fdef.name == 'map_func':
+                        inline_closure_call(f_ir, _globals, block, i, func)
+                        break
+
+        # remove sentinel global to avoid type inference issues
+        ir_utils.remove_dead(f_ir.blocks, f_ir.arg_names, f_ir)
+        f_ir._definitions = build_definitions(f_ir.blocks)
+        if use_nan:
```

Comment:
`arg_types` uses `series_var` for all argument types. It should use the types of `rhs.args[0]` and `rhs.args[2]`.